symengine/tests/basic/test_functions.cpp

```diff
@@ -4081,3 +4082,90 @@ TEST_CASE("test_dummy", "[Dummy]")
     CHECK(neq(*xdummy1, *x1));
     CHECK(neq(*xdummy1, *x1->as_dummy()));
 }
+
+TEST_CASE("test_sign", "[Sign]")
+{
+    RCP<const Symbol> x = symbol("x");
+    RCP<const Symbol> y = symbol("y");
+    RCP<const Basic> r = sign(one);
+    RCP<const Basic> s;
+    CHECK(eq(*r, *one));
+
+    r = sign(minus_one);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(zero);
+    CHECK(eq(*r, *zero));
+
+    r = sign(minus_one);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(real_double(1.2));
+    CHECK(eq(*r, *one));
+
+    r = sign(real_double(0.0));
+    CHECK(eq(*r, *zero));
+
+    r = sign(real_double(-1.2));
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(rational(-1, 2));
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(rational(1, 2));
+    CHECK(eq(*r, *one));
+
+    r = sign(rational(0, 2));
+    CHECK(eq(*r, *zero));
+
+    r = sign(Inf);
+    CHECK(eq(*r, *one));
+
+    r = sign(NegInf);
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(ComplexInf);
+    std::cout << *r << std::endl;
+    CHECK(r->__str__() == "sign(zoo)");
+
+    r = sign(Nan);
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *Nan));
+
+    r = sign(complex_double(std::complex<double>(0, 3)));
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *I));
+
+    r = sign(complex_double(std::complex<double>(0, -3)));
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *mul(I, minus_one)));
+
+    r = sign(pi);
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *one));
+
+    r = sign(E);
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *one));
+
+    r = sign(sign(x));
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *sign(x)));
+
+    r = sign(pow(I, rational(1, 2)));
+    std::cout << *r << std::endl;
+    CHECK(eq(*r, *I));
+
+    r = mul(mul(integer(2), x), pow(y, integer(3)));
+    std::cout << *r << std::endl;
+    s = sign(mul(x, pow(y, integer(3))));
+    CHECK(eq(*sign(r), *s));
+    CHECK(s->__str__() == "sign(x*y**3)");
+
+    r = mul(mul(mul(integer(2), x), pow(y, integer(3))), I);
+    std::cout << *r << std::endl;
+    s = mul(sign(mul(x, pow(y, integer(3)))), I);
+    CHECK(eq(*sign(r), *s));
+    CHECK(s->__str__() == "I*sign(x*y**3)");
```

Comment:
@isuruf This line seems to raise a canonicalization error, even though the tests pass locally. What's the reason behind that?